packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -482,7 +482,11 @@ function ChildReconciler(shouldTrackSideEffects) {
     newChild: any,
     expirationTime: ExpirationTime,
   ): Fiber | null {
-    if (typeof newChild === 'string' || typeof newChild === 'number') {
+    if (
+      typeof newChild === 'string' ||
+      typeof newChild === 'number' ||
+      (newChild && typeof newChild.valueOf() === 'string')
```

Comment:
It seems unlikely that we'd want to add the overhead of calling `valueOf` in this code path. Especially because `valueOf` can end up executing arbitrary user code.

Reply:
Yeah I'm certain we wouldn't accept this PR for this edge case. Would recommend instead to call toString() on the object in app code, or use regular strings. The issues with typescript, should probably be fixed at the types level, not in react.

Reply:
Sounds logical to me.

For reference, here is a much better string checker.
https://github.com/lodash/lodash/blob/master/isString.js

>Is there a way to just special case String so it doesn't hit reconcileChildrenIterator?

Probably possible, could be added in getIteratorFn?

```jsx
export function getIteratorFn(maybeIterable: ?any): ?() => ?Iterator<*> {
  if (maybeIterable === null || typeof maybeIterable !== 'object') {
    return null;
  }
  if (maybeIterator instanceof String) { // <--- ?
    return null;
  }
...
```